sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -307,20 +307,28 @@ def update_entity(  # pylint:disable=R1710
         row_key = entity['RowKey']
         entity = _add_entity_properties(entity)
         try:
+            metadata = None
             if mode is UpdateMode.REPLACE:
-                self._client.table.update_entity(
+                metadata, _ = self._client.table.update_entity(
                     table=self.table_name,
                     partition_key=partition_key,
                     row_key=row_key,
                     table_entity_properties=entity,
                     if_match=if_match or if_not_match or "*",
+                    cls=kwargs.pop('cls', _return_headers_and_deserialized),
                     **kwargs)
             elif mode is UpdateMode.MERGE:
-                self._client.table.merge_entity(table=self.table_name, partition_key=partition_key,
-                                                row_key=row_key, if_match=if_match or if_not_match or "*",
-                                                table_entity_properties=entity, **kwargs)
+                metadata, _ = self._client.table.merge_entity(
+                    table=self.table_name,
+                    partition_key=partition_key,
+                    row_key=row_key,
+                    if_match=if_match or if_not_match or "*",
+                    table_entity_properties=entity,
+                    cls=kwargs.pop('cls', _return_headers_and_deserialized),
```

Comment:
@annatisch had mentioned before that she prefers something more like 
`cls=kwargs.pop('cls', None) or _return_headers_and_deserialized` (can't remember this comment correctly, I think it might have had something to do with if the user passes in `None` for `cls`

Reply:
The way it currently is if the user passes in `None` that will be passed to the service.

Reply:
It wont be passed to the service - as this parameter deals with client-side behaviour.
But yeah... this one is tricky, because if a user _does_ provide a `cls` parameter, and it doesn't return a tuple of two values - the function will fail.....